gtk4/src/file_chooser.rs

```diff
@@ -0,0 +1,28 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::FileChooser;
+use glib::translate::*;
+use glib::IsA;
+
+pub trait FileChooserExtManual: 'static {
+    #[doc(alias = "gtk_file_chooser_add_choice")]
+    fn add_choice(&self, id: &str, label: &str, options: &[(Option<&str>, Option<&str>)]);
+}
+
+impl<O: IsA<FileChooser>> FileChooserExtManual for O {
+    fn add_choice(&self, id: &str, label: &str, options: &[(Option<&str>, Option<&str>)]) {
```

Comment:
I don't think these can be `None`? But you can pass `NULL` to both pointers.

Reply:
They can actually both be None. At least per the docs. The first item of the tuple is the option identifier, if it's set to None, the option is set as a boolean (checkbox) instead of a menu. 

I will give it a try in an example and see if everything works as expected :)

Reply:
That's not how I understand it. The whole thing can be `NULL`, or you have two string arrays of the same length. The string arrays are `NULL` terminated, so you can't have `NULL`s in the middle of them.